src/herbpy/action/actions.py

```diff
@@ -0,0 +1,145 @@
+import numpy, openravepy, prpy, time
+from prpy import tsr
+from prpy.tsr import tsrlibrary
+from prpy.tsr.tsr import *
+
+@ActionMethod
+def MoveObject(robot, direction = [0, 0, 1], distance = 0.1):
+	"""
+	@param robot The robot performing the push grasp
+	@param direction The direction to move the object
+	@param distance The distance to move the object
+	"""
+    try:
+        manip = robot.GetActiveManipulator()
+        manip.PlanToEndEffectorOffset(direction, distance, position_tolerance=0.1, execute=True)
+        return True
+    except Exception, e:
+        print 'MoveObject planning failed: ', str(e)
+        return False
+
+@ActionMethod
+def GraspHandle(robot, pitcher):
+    """
+    @param robot The robot performing the handle grasp
+    @param pitcher The pitcher who's handle will be grasped  
+    """
+    try:
+        # Get the pose of the pitcher
+        pitcher_in_world = pitcher.GetTransform()
+
+        # Get the AABB of the pitcher
+        pitcher_aabb = pitcher.ComputeAABB()
+
+        # spout in pitcher
+        spout_in_pitcher = numpy.array([[-0.7956, 0.6057, 0., -0.0662],
+                                        [-0.6057, -0.7956, 0., -0.0504],
+                                        [0., 0., 1., 0.2376], 
+                                        [0., 0., 0., 1.]])
+
+        # we want a hand pose orthogonal to the direction of the spout
+        spout_direction = numpy.arctan2(spout_in_pitcher[0,1], spout_in_pitcher[0,0])
+        palm_direction = spout_direction - 0.5*numpy.pi
+
+        ee_in_pitcher = numpy.eye(4)
+        ee_in_pitcher[:3,:3] = numpy.array([[0, 0, 1],
+                                            [-1, 0, 0],
+                                            [0, -1, 0]])
```

Comment:
This `numpy.array()` call creates an integer array. I suggest adding at least one float (e.g. `0.`) to avoid any unpleasant surprises with clipping to integers.

In this specific case it's probably fine because you immediately store the `array` in a float `array` - but I'd still prefer if we didn't risk. it :smile: